From 623ac82add409b3a8487fbb045e5262de37e2967 Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Tue, 16 Sep 2014 17:08:20 -0400
Subject: [usb] sierra: fix resume error reporting

Message-id: <1410887463-20674-429-git-send-email-dzickus@redhat.com>
Patchwork-id: 95567
O-Subject: [RHEL7 PATCH 428/591] USB: sierra: fix resume error reporting
Bugzilla: 1110939
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1110939

commit f4a2d499e7f0ea9089c8e537ddad12260f7aab69
Author: Johan Hovold <jhovold@gmail.com>
Date:   Mon May 26 19:22:57 2014 +0200

    USB: sierra: fix resume error reporting

    Add error message to resume error path and make sure to also return an
    error when failing to submit a cached write.

    Signed-off-by: Johan Hovold <jhovold@gmail.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/usb/serial/sierra.c b/drivers/usb/serial/sierra.c
index 96ad379..0254f6d 100644
--- a/drivers/usb/serial/sierra.c
+++ b/drivers/usb/serial/sierra.c
@@ -1037,6 +1037,10 @@ static int sierra_resume(struct usb_serial *serial)
    intfdata->in_flight++;
    err = usb_submit_urb(urb, GFP_ATOMIC);
    if (err < 0) {
+    dev_err(&port->dev,
+     "%s - submit urb failed: %d",
+     __func__, err);
+    ec++;
     intfdata->in_flight--;
     usb_unanchor_urb(urb);
     kfree(urb->transfer_buffer);
-- 
1.7.1